### PR TITLE
MohammadH/ In all-articles page scroll to the spot where the user was

### PR DESCRIPTION
**Pull request by mohammad-rework**
…on the post card when they come back to all-articles page

Changes:

-   In “All articles” when we select some article and land on the page, and click “back” to the [https://deriv.com/academy/blog/](url) page load from the top. Need to ensure the user can land back to the same spot they left.

## Type of change

-   [ ] Bug fix
-   [x] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

**Review by alamin-deriv — APPROVED** [review]

File: src/pages/academy/blog/index.tsx [hunk 1]
```diff
@@ -69,6 +69,38 @@ const ArticlesPage = ({ data }) => {
         og_description:
             'Educational content at your fingertips – everything you need to know to start trading or upgrade your trading skills.',
     }
+
+    const [scrollPosition, setScrollPosition] = React.useState(0)
```
Comment (prince-deriv) [inline src/pages/academy/blog/index.tsx:73]:
```suggestion
    const [scroll_position, setScrollPosition] = React.useState(0)
```

File: src/pages/academy/blog/index.tsx [hunk 2]
```diff
@@ -69,6 +69,38 @@ const ArticlesPage = ({ data }) => {
...
+    const handleScroll = () => {
+        const position = window.pageYOffset
+        setScrollPosition(position)
+    }
+
+    React.useEffect(() => {
```
Comment (prince-deriv) [inline src/pages/academy/blog/index.tsx:79]: import `useEffect` from react 

```suggestion
   useEffect(() => {
```

File: src/pages/academy/blog/index.tsx [hunk 3]
```diff
@@ -69,6 +69,38 @@ const ArticlesPage = ({ data }) => {
         og_description:
             'Educational content at your fingertips – everything you need to know to start trading or upgrade your trading skills.',
     }
+
+    const [scrollPosition, setScrollPosition] = React.useState(0)
+    const handleScroll = () => {
+        const position = window.pageYOffset
+        setScrollPosition(position)
```
Comment (salarhfz-fs) [inline src/pages/academy/blog/index.tsx:76]:
```suggestion
        setScrollPosition(window.pageYOffset)
```

File: src/pages/academy/blog/index.tsx [hunk 4]
```diff
@@ -69,6 +69,38 @@ const ArticlesPage = ({ data }) => {
...
+            window.removeEventListener('scroll', handleScroll)
+        }
+    }, [])
+
+    React.useEffect(() => {
+        scrollPosition && localStorage.setItem('scroll', scrollPosition.toString())
+    }, [scrollPosition])
+
+    React.useEffect(() => {
+        const prevScroll = parseInt(localStorage.getItem('scroll'))
```
Comment (salarhfz-fs) [inline src/pages/academy/blog/index.tsx:91]: I think if `localStorage.getItem('scroll')` return `null` then becomes `NaN`. What do you think?

Comment (mohammad-rework) [reply src/pages/academy/blog/index.tsx:91]: You are right but I think it never returns `null` due to `useEffect` on line 79. What do you think?

Comment (salarhfz-fs) [reply src/pages/academy/blog/index.tsx:91]: I think there are chances. In the first page load for example.
We can simple prevent hat by adding `prevScroll !== NaN` to the condition.
BTW I'm not sure 100%, just a suggestion.

**Review by salarhfz-fs — APPROVED** [review]

**Pull request closed without merging**